core/src/main/java/com/linecorp/armeria/server/DecoratingService.java

```diff
@@ -65,6 +66,11 @@ public void serviceAdded(ServiceConfig cfg) throws Exception {
         return result.isPresent() ? result : delegate.as(serviceType);
     }
 
+    @Override
+    public boolean shouldCachePath(PathAndQuery pathAndQuery, PathMapping pathMapping) {
```

Comment:
`PathAndQuery` is an internal class. Could you replace it with two `String` parameters?